core/src/css/test/theme-ionic/typography.e2e.ts

```diff
@@ -0,0 +1,185 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+/**
+ * Text does not have per-mode styles
+ */
+configs({ directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('typography: global styles'), () => {
+    test('should apply the globals styles for each semantic tag', async ({ page }) => {
+      await page.setContent(
+        `
+        <link href="../../../../../css/ionic.globals.ionic.css" rel="stylesheet" />
+        <style>
+          .ionic-preview {
+            background: #d7d8da;
+            border-radius: 16px;
+            box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+            font-size: 16px;
+            font-weight: 100;
+            margin-bottom: 8px;
+            padding: 8px;
+          }
+        </style>
+
+        <ion-content class="ion-padding" id="content">
+          <ion-text class="ionic-preview">HTML Tags</ion-text>
+          <p>
+            <h1>H1: The quick brown fox jumps over the lazy dog</h1>
+            <h2>H2: The quick brown fox jumps over the lazy dog</h2>
+            <h3>H3: The quick brown fox jumps over the lazy dog</h3>
+            <h4>H4: The quick brown fox jumps over the lazy dog</h4>
+            <h5>H5: The quick brown fox jumps over the lazy dog</h5>
+            <h6>H6: The quick brown fox jumps over the lazy dog</h6>
+          <ion-text class="ionic-font-size-l">
+            <p>Tag_P FontLarge: The quick brown fox jumps over the lazy dog</p>
+          </ion-text>
+          <p>
+            <ion-text class="ionic-font-size-s">
+              <span>Tag_Span FontSmall: The quick brown fox jumps over the lazy dog</span>
+            </ion-text>
+          </p>
+          <p>
+            <ion-text class="ionic-font-size-base">Tag_None FontBase: The quick brown fox jumps over the lazy
+              dog</ion-text>
+          </p>
+          </p>
+        </ion-content>
+      `,
+        config
+      );
+
+      const globalStyles = page.locator('ion-content');
+      await expect(globalStyles.nth(0)).toHaveScreenshot(screenshot(`ionic-semantic-global-styles`));
```

Comment:
I used the example that is present on text styles tests, maybe for the typography test doesn't make sense to use it. I checked the other tests on `css -> test -> a11y` and used the approach of adding a `div` element to wrapped the elements.

Reply:
Ah, in that case, the `nth(0)` is being used because the test has multiple `ion-text`. The test only wants to capture the very first one so everything is in the snapshot, just like how you capture `div` in your test.